apollo-federation/src/sources/connect/id.rs

```diff
@@ -119,4 +109,78 @@ impl ConnectorPosition {
             ConnectorPosition::Type(pos) => format!("{}_{}", pos.type_name, pos.directive_index),
         }
     }
+
+    pub(super) fn on_query(&self, schema: &Schema) -> bool {
+        schema
+            .schema_definition
+            .query
+            .as_ref()
+            .map(|query| match self {
+                ConnectorPosition::Field(pos) => *pos.field.type_name() == query.name,
+                ConnectorPosition::Type(_) => false,
+            })
+            .unwrap_or_default()
+    }
+
+    pub(super) fn on_mutation(&self, schema: &Schema) -> bool {
+        schema
+            .schema_definition
+            .mutation
+            .as_ref()
+            .map(|mutation| match self {
+                ConnectorPosition::Field(pos) => *pos.field.type_name() == mutation.name,
+                ConnectorPosition::Type(_) => false,
+            })
+            .unwrap_or_default()
+    }
+}
+
+/// Reifies the connector position into schema definitions
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) enum ConnectedElement<'schema> {
+    Field {
+        parent_type: &'schema ExtendedType,
+        field_def: &'schema Component<FieldDefinition>,
+    },
+    Type {
+        type_def: &'schema ExtendedType,
+    },
+}
+
+impl Display for ConnectedElement<'_> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Self::Field {
+                parent_type,
+                field_def,
+            } => write!(f, "{}.{}", parent_type.name(), field_def.name),
+            Self::Type { type_def } => write!(f, "{}", type_def.name()),
+        }
+    }
+}
+
+impl ConnectedElement<'_> {
+    pub(super) fn on_query(&self, schema: &Schema) -> bool {
+        schema
+            .schema_definition
+            .query
+            .as_ref()
+            .map(|query| match self {
+                ConnectedElement::Field { parent_type, .. } => *parent_type.name() == query.name,
+                ConnectedElement::Type { .. } => false,
+            })
+            .unwrap_or_default()
+    }
+
+    pub(super) fn on_mutation(&self, schema: &Schema) -> bool {
+        schema
+            .schema_definition
+            .mutation
+            .as_ref()
+            .map(|mutation| match self {
+                ConnectedElement::Field { parent_type, .. } => *parent_type.name() == mutation.name,
+                ConnectedElement::Type { .. } => todo!(),
```

Comment:
```suggestion
                ConnectedElement::Type { .. } => false,
```

Also, looks like `on_query` and `on_mutation` are only used to check `on_root_type`, so should they be a single function to reduce code?